src/scripts/main.js

```diff
@@ -1,11 +1,30 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve) => {
+    const handleEvent = () => {
+      resolve(
+        `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}`,
+      );
+      element.removeEventListener(eventName, handleEvent);
```

Comment:
You are removing the event listener immediately after the event is fired for the first time. This means that the `waitFor` function will only work once for each element-event pair. If this is the intended behavior, then it's fine. Otherwise, you might want to consider removing this line.